_posts/2020-03-19-our-rails-upgrade-process.markdown

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title: "Our Rails Upgrade Process"
+date: 2020-03-17 10:30:00
+reviewed: 2020-03-19 10:00:00
+categories: ["rails", "upgrade"]
+author: "cleiviane"
+---
+
+We know that there are many challenges involving a Rails upgrade project. Depending on how big is your application, how old is your Rails version and how well structured your code is, it can be difficult to perform that job and keep your sanity. If you don't find a reliable and trustable process to guide you from version X to version Y, you can end-up in a nightmare.
+
+The good news is that here at [Ombu Labs](https://www.ombulabs.com/), we have been upgrading Rails applications for over 10 years now and this gave us a know-how to define a process that has proven to be very effective. And today I want to share Our Rails Upgrade Process with you.
+
+<!--more-->
+
+## Step 1: Finding deprecation warnings
+
+The first thing that we do is to search for deprecation warnings in the application, which are log messages suggesting things that you will need to change when moving for the next version of Rails. That can be because something will either be removed from the source code or the code will work in a different way.
+
+You might find these deprecation warnings by searching at the `production.log` or tracking then down in your Log Management Software, such as [Sentry](https://sentry.io/) or [Splunk](https://www.splunk.com).
+
+For each different deprecation warning that we find, we create one story in our backlog software (at Ombu Labs we use [Pivotal Tracker](pivotaltracker.com) to help us with that). This will make things easier to code review and to test in QA.
+
+## Step 2: Fixing all deprecation warnings
+
+After create all stories for the deprecation warnings, is time to address them one by one. Deprecation warnings are usually pretty straightforward, the messages are clear about what needs to change. They are also backwards compatible which means that you can fix them and merge directly to the current version of Rails.
+
+A good examples is the deprecation warnings caused by the changes in `ActiveRecord::Dirty` in the Rails 5.2 version. We wrote [this article](https://www.fastruby.io/blog/rails/upgrades/active-record-5-1-api-changes.html) while ago showing how to address them.
+
```

Comment:
Ideally, you will have a test suite that exercises the code that you are changing. Then you can be sure that the changes don’t break existing behavior. If you don’t have tests, you will have to manually execute the code that you’re changing. That is when it can become a tedious process.